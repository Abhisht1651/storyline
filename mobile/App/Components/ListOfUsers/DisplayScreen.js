import React from 'react'
import { View } from 'react-native'
import ListOfUsers from './ListOfUsers'
import styles from '../Styles/ListOfUserStyles/DisplayScreenStyle'

const DisplayScreen = () => {
  return (
    <View style={styles.container}>
      <ListOfUsers name="Name" optionalText="Optional Text" />
      <ListOfUsers type={2} buttonText="Message" name="Amelia Jones" optionalText="Optional Text" />
    </View>
  )
}

export default DisplayScreen
