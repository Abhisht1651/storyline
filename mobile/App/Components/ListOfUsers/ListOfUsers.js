// @flow

import React, { useState, useEffect } from 'react'
import { View, Text, TouchableOpacity, ImageProps } from 'react-native'
import PropTypes from 'prop-types'
import styles from '../Styles/ListOfUserStyles/ListOfUsersStyle'
import images from '../../Themes/Images'
import MediumProfilePicture from '../ProfilePicture/MediumProfilePictures/MediumProfilePicture'

type Props = {
  name: string,
  optionalText: string,
  image: ImageProps,
  buttonText: string,
  buttonTextAfterPress: string,
  type: number,
  onPress: PropTypes.func,
  profileType: number,
  onCancelPress: PropTypes.func
}

const ListOfUsers = (props: Props) => {
  const {
    name,
    optionalText,
    image,
    buttonText,
    buttonTextAfterPress,
    type,
    onPress,
    profileType,
    onCancelPress
  } = props

  const [buttonPressed, setButtonPressed] = useState(false)
  const [buttonStyle, setButtonStyle] = useState({ borderColor: '#7B46E4' })
  const [textStyle, setTextStyle] = useState({ color: '#7B46E4' })
  const [btnText, setBtnText] = useState(buttonText)

  useEffect(() => {
    if (buttonPressed) {
      setButtonStyle({ borderColor: '#DCDCE9' })
      setTextStyle({ color: '#8F8FAF' })
      setBtnText(buttonTextAfterPress)
    } else {
      setButtonStyle({ borderColor: '#7B46E4' })
      setTextStyle({ color: '#7B46E4' })
      setBtnText(buttonText)
    }
  }, [buttonPressed])

  const onPressHandler = () => {
    setButtonPressed(!buttonPressed)
    if (buttonPressed) onCancelPress()
    else if (!buttonPressed) onPress()
  }

  return (
    <View style={styles.container}>
      <View style={styles.profileContainer}>
        <TouchableOpacity style={styles.imageContainer} activeOpacity={1}>
          <MediumProfilePicture type={profileType} image={image} />
        </TouchableOpacity>
        <View style={styles.textBlock}>
          <Text style={styles.nameText}>{name}</Text>
          {optionalText && type === 1 ? (
            <Text style={styles.optionalText}>{optionalText}</Text>
          ) : null}
        </View>
      </View>
      <View style={type === 1 ? styles.btnContainer1 : styles.btnContainer2}>
        <TouchableOpacity
          activeOpacity={1}
          style={[styles.button, buttonStyle]}
          onPress={type === 1 ? onPressHandler : onPress}>
          <Text style={[styles.buttonText, textStyle]}>{btnText}</Text>
        </TouchableOpacity>
      </View>
    </View>
  )
}

ListOfUsers.defaultProps = {
  name: 'Name',
  optionalText: 'Optional Text',
  image: images.voterIcon2,
  buttonText: 'Add',
  buttonTextAfterPress: 'Added',
  type: 1,
  profileType: 1,
  onPress: () => {},
  onCancelPress: () => {}
}

export default ListOfUsers
