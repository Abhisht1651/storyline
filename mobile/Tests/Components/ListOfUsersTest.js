import { shallow } from 'enzyme'
import React from 'react'

import ListOfUsers from '../../App/Components/ListOfUsers/ListOfUsers'

describe('ListOfUsers', () => {
  it('should match to snapshot', () => {
    const component = shallow(<ListOfUsers />)
    expect(component).toMatchSnapshot()
  })

  it('should render Text ', () => {
    const wrapper = shallow(<ListOfUsers />)
    expect(wrapper.find('Text')).toHaveLength(3)
  })

  it('should render Text for type 2 ', () => {
    const wrapper = shallow(<ListOfUsers type={2} />)
    expect(wrapper.find('Text')).toHaveLength(2)
  })
})
