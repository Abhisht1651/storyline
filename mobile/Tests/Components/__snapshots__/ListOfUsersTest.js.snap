// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfUsers should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomWidth": 2,
      "borderColor": "#E6ECED",
      "flexDirection": "row",
      "height": 152,
      "justifyContent": "space-between",
      "width": 652,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <ForwardRef
      activeOpacity={1}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#DCDCE9",
          "borderRadius": 44,
          "height": 88,
          "justifyContent": "center",
          "marginRight": 26,
          "width": 88,
        }
      }
    >
      <MediumProfilePicture
        image={
          Object {
            "default": "",
          }
        }
        onPress={[Function]}
        type={1}
      />
    </ForwardRef>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "marginRight": 26,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Medium",
            "fontSize": 36,
            "letterSpacing": -0.12,
          }
        }
      >
        Name
      </Text>
      <Text
        style={
          Object {
            "color": "#8F8FAF",
            "fontFamily": "BananaGrotesk-Regular",
            "fontSize": 24,
            "letterSpacing": -0.07,
          }
        }
      >
        Optional Text
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "justifyContent": "center",
        "width": 150,
      }
    }
  >
    <ForwardRef
      activeOpacity={1}
      onPress={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#7B46E4",
            "borderRadius": 32,
            "borderWidth": 2,
            "height": 60,
            "justifyContent": "center",
            "width": "100%",
          },
          Object {
            "borderColor": "#7B46E4",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#7B46E4",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 28,
              "letterSpacing": -0.08,
            },
            Object {
              "color": "#7B46E4",
            },
          ]
        }
      >
        Add
      </Text>
    </ForwardRef>
  </View>
</View>
`;
